ts/sagas/startup/__tests__/loadServiceContentRequestHandler.test.ts

```diff
@@ -0,0 +1,126 @@
+import { right } from "fp-ts/lib/Either";
+import * as pot from "italia-ts-commons/lib/pot";
+import {
+  NonEmptyString,
+  OrganizationFiscalCode
+} from "italia-ts-commons/lib/strings";
+import { testSaga } from "redux-saga-test-plan";
+import { DepartmentName } from "../../../../definitions/backend/DepartmentName";
+import { OrganizationName } from "../../../../definitions/backend/OrganizationName";
+import { ServiceId } from "../../../../definitions/backend/ServiceId";
+import { ServiceName } from "../../../../definitions/backend/ServiceName";
+import { ServicePublic } from "../../../../definitions/backend/ServicePublic";
+import { updateOrganizations } from "../../../store/actions/organizations";
+import { loadServiceContent } from "../../../store/actions/services";
+import {
+  organizationNamesByFiscalCodeSelector,
+  OrganizationNamesByFiscalCodeState
+} from "../../../store/reducers/entities/organizations/organizationsByFiscalCodeReducer";
+import { visibleServicesSelector } from "../../../store/reducers/entities/services/visibleServices";
+import { loadServiceContentRequestHandler } from "../loadServiceContentRequestHandler";
+
+const mockedServiceId = "A01";
+const getService = jest.fn();
+const mockedAction = {
+  type: loadServiceContent.request,
+  payload: mockedServiceId
+};
+const mockedService: ServicePublic = {
+  service_id: mockedServiceId as ServiceId,
+  service_name: "servizio1" as ServiceName,
+  organization_name: "ente1 - nuovo nome" as OrganizationName,
+  department_name: "dipartimento1" as DepartmentName,
+  organization_fiscal_code: "01" as OrganizationFiscalCode,
+  version: 2
+};
+const mockedGenericError = new Error(`response status 500`);
+
+const mockedVisibleServices = pot.some([
+  { service_id: "A01" as ServiceId, version: 1 },
+  { service_id: "A02" as ServiceId, version: 5 },
+  { service_id: "A03" as ServiceId, version: 2 }
+]);
+
+describe("loadServiceContentRequestHandler", () => {
+  it("returns an error if backend response is 500", () => {
+    testSaga(loadServiceContentRequestHandler, getService, mockedAction)
+      .next()
+      .call(getService, { service_id: mockedServiceId })
+      .next(right({ status: 500, value: "generic error" }))
+      .put(
+        loadServiceContent.failure({
+          service_id: mockedServiceId,
+          error: mockedGenericError
+        })
+      )
+      .next()
+      .isDone();
+  });
+
+  it("returns service content if the backand response is 200", () => {
+    testSaga(loadServiceContentRequestHandler, getService, mockedAction)
+      .next()
+      .call(getService, { service_id: mockedServiceId })
+      .next(right({ status: 200, value: mockedService }))
+      .put(loadServiceContent.success(mockedService))
+      .next();
```

Comment:
```suggestion
      .next() // update organization name
      .next()
      .isDone();
```
in this way you can skip the update organization name logic but you can check that the generator ends successfully

Reply:
To improve the implementation, I moved the code about the organization name check into a dedicated function (https://github.com/teamdigitale/io-app/pull/1503/commits/a1f5fafde768f31ddf2c6f23126aa5ce0772268a), updating the existing `loadServiceContentRequestHandler.test.ts` and adding the test also for the new function.